src/core/server/http/router/validator/validator.ts

```diff
@@ -274,7 +274,7 @@ export class RouteValidator<P = {}, Q = {}, B = {}> {
       // if options.body.output === 'stream'
       return schema.stream();
     } else {
-      return schema.maybe(schema.nullable(schema.object({}, { allowUnknowns: true })));
+      return schema.maybe(schema.nullable(schema.any({})));
     }
   }
```

Comment:
I'm not really sure what the purpose of the `preValidateSchema` method really is. It data is a buffer or stream, we return the associated schema which is assured to validate. If data is 'plain' javascript data, we validates with the most permissive schema we can. Either way, the validation is assured. As the security-related validation are supposed to be done in `safetyPrechecks` and `safetyPostchecks` anyway, I have the feeling this method should be removed?

@afharo thoughts ?

Reply:
After talks with the security team, we want to have a minimum validation on our end. 

On top of that, according to the types, we only allow `Buffer`, `Stream` or `object` in the body. So the goal of `preValidateSchema` is to ensure any of those.

In the route `options`, [the `parse` property](https://github.com/elastic/kibana/blob/4d659477ad5993d672cfe04384f259ce0e9e65f7/src/core/server/http/router/route.ts#L85-L94) uses Hapi to convert any incoming request into an object, when possible if set to `true`. When set to `false` or `'gunzip'`, it will return a `Buffer`. So, when is it not possible to convert to object, based on the [known `Content-Type`s](https://github.com/elastic/kibana/blob/4d659477ad5993d672cfe04384f259ce0e9e65f7/src/core/server/http/router/route.ts#L55-L63)? Maybe for `text/*`?

Reply:
> if the request 'Content-Type' matches the allowed mime types set by allow (for the whole payload as well as parts), the payload is converted into an object when possible.

It doesn't mean the body **always** is an object.

>  So, when is it not possible to convert to object, based on the known Content-Types? 

For example, `array` is a valid JSON. Although validator throws on it https://github.com/elastic/kibana/issues/54268

I also don't understand why we need a pre-validation step. At the end of the day, the `customValidation` is the only source of truth for the validator.